query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_7434.rs

```diff
@@ -0,0 +1,20 @@
+use query_engine_tests::*;
+
+#[test_suite(schema(autoinc_id), capabilities(CreateMany, AutoIncrement))]
+mod not_in_batching {
+    use query_engine_tests::Runner;
+
+    #[connector_test]
+    async fn not_in_batch_filter(runner: Runner) -> TestResult<()> {
+        runner.query(r#"mutation { createManyTestModel(data: [{},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{}]) { count }}"#).await?.assert_success();
+
+        assert_error!(
+            runner,
+            "query { findManyTestModel(where: { id: { notIn: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11] } }) { id }}",
+            2029,
+            "Your query cannot be split into multiple queries because the filter will cause incorrect results"
```

Comment:
While this error makes sense for us, I would be greatly confused as a user. Since this is a user-facing error with a code I propose we package it into something that explains more and a call to action to possibly solve that.

One example (that isn't necessarily what you should take):
"Parameter limits for this database provider require to split the query into one or more parts, but negation filters prevent Prisma from splitting. Please try reducing the used values in the query."

Reply:
What about:

```
"Parameter limits for this database provider require this query to be split into multiple queries, but the negation filters used prevent the query from being split. Please reduce the used values in the query."
```